soc/arm/arm/mps2/soc.c

```diff
@@ -33,8 +33,20 @@ FPGAIO_INIT(2);
 #define SSE_200_SYSTEM_CTRL_CPU_WAIT	(SSE_200_SYSTEM_CTRL_S_BASE + 0x118)
 #define SSE_200_CPU_ID_UNIT_BASE	(0x5001F000UL)
 
+/* The starting address for the application image on the secondary
```

Comment:
@microbuilder @galak i still cannot  imeediately understand which of the 2 is correct:
- there is a second CPU on AN521, which just does not have TrustZone and is by default non-secure
- you refer to the non-secure image address offsets of the CPU we build Zephyr on.

Reply:
here you are using "non secure cpu", later, "non secure cpu image"

Reply:
It's the base address that the application image running on the secondary mcu starts at. It seems clear to me that an image runs on an mcu, and it starts at a specific address??

If you want to propose something specific, I'm happy to change it.

Reply:
so i understand that there are 3 images you are talking about:
- tf-m runs as the SPE on primary CPU
- Zephyr runs as the NSPE on primary CPU
- and there is another firmware running on a secondary CPU

correct @microbuilder ?

Reply:
```
/* The base address tha the application image will start at on the secondary
 * (non-TrustZone) Cortex-M33 mcu.
 */
#define NON_SECURE_FLASH_ADDRESS	(0x100000)

/* The memory map offset for the application image, which is used
 * to determine the location of the reset vector at startup.
 */
#define NON_SECURE_FLASH_OFFSET	        (0x10000000)

/* Space reserved for TF-M's secure bootloader on the secondary mcu.
 * This space is reserved whether BL2 is used or not.
 */
#define BL2_HEADER_SIZE                 (0x400)
```

We can clarify this with ARM or dig into the DS to change this later, but seems silly to hold up the TF-M merge tonight for comments on fields that aren't used, even if I agree clear documentation is important?

Reply:
@microbuilder we are not at all holding up the merge - just want things to look nice.


So, since we now clarified that this is related to an image of a second Cortex-M33 on AN521, that is non-TrustZone enabled, I suggest you rename the macros, and remove the NONSECURE prefix, because this made me think the offsets are related to the non-secure NSPE

Reply:
>     * and there is another firmware running on a secondary CPU, which happens to be non-secure

The first two are right, yes.

Nothing is running on the secondary CPU, though. This code for the second CPU is never used by the TF-M demos, but seems to have been added for the following demos:

```
samples/subsys/ipc/ipm_mhu_dual_core/src/main.c:65:          wakeup_cpu1();
samples/subsys/ipc/openamp/src/main.c:294:   wakeup_cpu1();
```

Reply:
> Nothing is running on the secondary CPU, though. This code for the second CPU is never used by the TF-M demos, but seems to have been added for the following demos:

Ok thanks a lot for clarifying. In any case, I believe you should remove the NONSECURE prefix from these local macros, because it is confusing; I thought they refer to the Zephyr NSPE.
The documentation you added is ok.

Reply:
Renamed values to `CPU1_*` and tested the two IPC samples, and seems OK, and agree NONSECURE is confusing.